pytng/pytng.pyx

```diff
@@ -1016,334 +1138,340 @@ cdef class TNGFileIterator:
-    cdef int64_t block_id
-    cdef bint debug
+    cpdef  get_time(self):
+        """Get the time of the current integrator step being read from the file
 
-    cdef int64_t step
-    cdef double frame_time
-    cdef double precision
-    cdef int64_t n_values_per_frame, n_atoms
-    cdef char[TNG_MAX_STR_LEN] block_name
-    cdef MemoryWrapper _wrapper  # manages the numpy array lifetime
-    cdef tng_function_status read_stat
-    cdef np.ndarray values  # the final values as a numpy array
+        Returns
+        -------
+        time : int
+            the time of the current step
+        """
+        cdef tng_function_status read_stat
+        cdef double _step_time
+        read_stat = self._get_step_time( & _step_time)
+        if read_stat != TNG_SUCCESS:
+            return None
+        else:
+            return _step_time
 
-    cdef bint block_is_read
+    cpdef np.ndarray get_positions(self, np.ndarray data):
+        """Get the positions present at the current step and read them into a
+        NumPy array
 
-    def __cinit__(self, TrajectoryWrapper traj, int64_t frame,
-                  bint debug=False):
+        Parameters
+        ----------
+        data : np.ndarray
+           NumPy array to read the data into. As this is a particle dependent
+           block, the shape should be (n_atoms, n_values_per_frame)
+           ie (n_atoms, 3).
+        """
+        self.get_blockid(TNG_TRAJ_POSITIONS, data)
+        return data
 
-        self._traj = traj._ptr
-        self._frame = frame
-        self.debug = debug
-        self.block_id = -1
+    cpdef np.ndarray get_box(self, np.ndarray data):
+        """Get the box vectors present at the current step and read them into a
+        NumPy array. The box vectors are a (3,3) matrix comprised of 3
+        three-dimensional basis vectors for the coordinate system of the
+        simulation. The vectors can be accessed in their proper shape by
+        reshaping the resulting (1,9) matrix to be (3,3) with
+        ndarray.reshape(3,3).
 
-        self.step = -1
-        self.frame_time = -1
-        self.precision = -1
-        self.n_values_per_frame = -1
-        self.n_atoms = -1
-        self.block_is_read = False
+        Parameters
+        ----------
+        data : np.ndarray
+           NumPy array to read the data into. As this is NOT a particle 
+           dependent block, the shape should be
+           (1, n_values_per_frame) ie (1,9)
+        """
+        self.get_blockid(TNG_TRAJ_BOX_SHAPE, data)
+        return data
 
-        # TODO allocs a single byte,
-        # this can be changed if the signature of MemoryWrapper is changed
-        self._wrapper = MemoryWrapper(1)
+    cpdef np.ndarray get_velocities(self, np.ndarray data):
+        """Get the velocities present at the current step and read them into a
+        NumPy array
 
-    def __dealloc__(self):
-        pass
+        Parameters
+        ----------
+        data : np.ndarray
```

Comment:
As I was looking through the tests I was wondering what the supported dtypes for these input arrays might be, and if an exception would be raised for i.e., non-numeric dtypes, etc.

Looking farther down below--is this all routed through the dtype check in `get_blockid()` then?

Reply:
I have tried to clarify available dtypes in the docs.

Yes you are right it all passes through that dtype check, and an exception is raised if incorrect.